### PR TITLE
Fix compiler crash for table multi-key-expression 

### LINKED ISSUE #35852: Bad sad for table member access with multiple non-existing keys
**Description:**
```ballerina
    table<Group> key(id) t1 = table [
            {
                id: 1,
                name: "a",
                area: "area1",
                age: "10",
                status: ""
            },
            {
                id: 2,
                name: "b",
                area: "area2",
                age: "20",
                status: ""
            }
        ];
    Group? g = t1[0, ""];
```

above will give the following.

```
[2022-04-19 22:35:29,827] SEVERE {b7a.log.crash} - class org.wso2.ballerinalang.compiler.semantics.model.types.BType cannot be cast to class org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType (org.wso2.ballerinalang.compiler.semantics.model.types.BType and org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType are in unnamed module of loader 'app') 
java.lang.ClassCastException: class org.wso2.ballerinalang.compiler.semantics.model.types.BType cannot be cast to class org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType (org.wso2.ballerinalang.compiler.semantics.model.types.BType and org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType are in unnamed module of loader 'app')
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkIndexAccessExpr(TypeChecker.java:7955)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:2893)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangIndexBasedAccess.accept(BLangIndexBasedAccess.java:60)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1052)
        at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4057)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4028)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:4020)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1907)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:50)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4057)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4028)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:4024)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:524)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4057)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:500)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4057)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:4028)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:4020)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:349)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:309)
        at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:188)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:114)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:436)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:386)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:213)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:198)
        at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:114)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:109)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:233)
        at io.ballerina.projects.Package.getCompilation(Package.java:145)
        at io.ballerina.projects.Package.runCodeGeneratorPlugins(Package.java:250)
        at io.ballerina.cli.task.CompileTask.execute(CompileTask.java:84)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:249)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:51)
 
ERROR [statements:(1:1,1:1)] Compilation failed due to: class org.wso2.ballerinalang.compiler.semantics.model.types.BType cannot be cast to class org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType (org.wso2.ballerinalang.compiler.semantics.model.types.BType and org.wso2.ballerinalang.compiler.semantics.model.types.BTupleType are in unnamed module of loader 'app')
```

**Affected Versions:**
Ballerina 2201.0.2 (Swan Lake)
Language specification 2022R1

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose

Fixes #35852

## Approach
According to spec, `A multi-key-expression E1, E2,..., En is equivalent to a list-constructor-expr [E1, E2,..., En].`
This PR models it as `BLangListConstructorExpr` and removes `BLangTableMultiKeyExpr`

### CODE REVIEW

**Comment by MaryamZi** [comment]
@KavinduZoysa, please resolve the conflicts.

File: tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal [hunk 1]
```diff
@@ -433,3 +433,56 @@ function testAssigningKeyedToEmptyKeyedTbl() {
...
+            name: "a"
+        },
+        {
+            id: 2,
+            name: "a"
+        }
+    ];
+
+    Person2? _ = t2[11, 12, 13, 14]; // error
+}
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal:488]: Shall we add a test with key-type-constraint as well

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 2]
```diff
@@ -8262,33 +8259,9 @@ private BType checkIndexAccessExpr(BLangIndexBasedAccess indexBasedAccessExpr, A
...
-                    dlog.error(indexBasedAccessExpr.pos, DiagnosticErrorCode.INVALID_KEY_CONSTRAINT_PROVIDED_FOR_ACCESS,
-                            keyTypeConstraint);
-                    return symTable.semanticError;
-                }
-            } else {
-                List<BLangExpression> multiKeyExpressionList = ((BLangTableMultiKeyExpr)
-                        indexBasedAccessExpr.indexExpr).multiKeyIndexExprs;
-                List<BType> keyConstraintTypes = ((BTupleType) keyTypeConstraint).tupleTypes;
-                if (keyConstraintTypes.size() != multiKeyExpressionList.size()) {
-                    dlog.error(indexBasedAccessExpr.pos, DiagnosticErrorCode.INVALID_KEY_CONSTRAINT_PROVIDED_FOR_ACCESS,
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:8277]: Should be able to remove `INVALID_KEY_CONSTRAINT_PROVIDED_FOR_ACCESS ` diagnostic as well

File: tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal [hunk 3]
```diff
@@ -433,3 +433,56 @@ function testAssigningKeyedToEmptyKeyedTbl() {
...
+    Person1? _ = t1[1, 2, 3]; // error
+}
+
+type Person2 record {
+    readonly int id;
+    readonly string name;
+};
+
+function testMultipleKeys2() {
+    table<Person2> key(id) t1 = table [
```
Comment (pcnfernando) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal:463]: Shall we refactor mappings in-line?

Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal:347]: Changed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java (hunk 2 above)
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:8249]: Removed

File: tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal (hunk 1 above)
Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal:360]: Added

**Review by pcnfernando — APPROVED** [review]
LGTM

**Review by hasithaa — APPROVED** [review]

**Pull request merged**